src/OpenCensus/Trace/Propagation/TraceContextFormat.cs

```diff
@@ -0,0 +1,136 @@
+
+            // TODO: validate span delimeter
+            var spanId = SpanId.FromBytes(Arrays.StringToByteArray(traceparent, "00-0af7651916cd43dd8448eb211c80319c-".Length, "00f067aa0ba902b7".Length));
+
+            // TODO: validate options delimeter
+            var optionsArray = Arrays.StringToByteArray(traceparent, "00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-".Length, 2);
+
+            var options = TraceOptions.Default;
+            if ((optionsArray[0] | 1) == 1)
+            {
+                options = TraceOptions.Builder().SetIsSampled(true).Build();
+            }
+
+            List<KeyValuePair<string, string>> entries = new List<KeyValuePair<string, string>>();
+            if (tracestateCollection != null)
+            {
+                foreach (var tracestate in tracestateCollection)
+                {
+                    // tracestate: rojo=00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01,congo=BleGNlZWRzIHRohbCBwbGVhc3VyZS4
+                    var keyStartIdx = 0;
+                    var length = tracestate.Length;
+                    while (keyStartIdx < length)
+                    {
+                        var keyEndIdx = tracestate.IndexOf("=", keyStartIdx);
+
+                        if (keyEndIdx == -1)
+                        {
+                            // error happened. Ignore this tracestate
+                            break;
+                        }
+
+                        var valueStartIdx = keyEndIdx + 1;
+
+                        var valueEndIdx = tracestate.IndexOf(",", valueStartIdx);
+                        valueEndIdx = valueEndIdx == -1 ? length : valueEndIdx;
+                        entries.Add(
+                            new KeyValuePair<string, string>(
+                                tracestate.Substring(keyStartIdx, keyEndIdx - keyStartIdx).Trim(), 
+                                tracestate.Substring(valueStartIdx, valueEndIdx - valueStartIdx).Trim()));
+                        keyStartIdx = valueEndIdx + 1;
+                    }
+                }
+            }
+
+            var tracestateBuilder = Tracestate.Builder;
+
+            entries.Reverse();
+            foreach (var entry in entries)
+            {
+                tracestateBuilder.Set(entry.Key, entry.Value);
+            }
+
+            return SpanContext.Create(traceId, spanId, options, tracestateBuilder.Build());
+        }
+
+        /// <inheritdoc/>
+        public override void Inject<T>(ISpanContext spanContext, T carrier, Action<T, string, string> setter)
+        {
+            var sb = new StringBuilder();
+            sb.Append("00-").Append(spanContext.TraceId.ToLowerBase16());
+            sb.Append('-').Append(spanContext.SpanId.ToLowerBase16());
+            sb.Append('-').Append(spanContext.TraceOptions.IsSampled ? "01" : "00");
+            setter(carrier, "traceparent", sb.ToString());
+
+            sb.Clear();
```

Comment:
when you need to concat a few strings, StringBuilder is not efficient. string.concat is the most efficient way to do so.  Please use it for traceparent

Reply:
ok. I'll keep string builder fro tracestate.

THis calls for https://github.com/census-instrumentation/opencensus-csharp/issues/15 Good first candidate